src/snowflake/snowpark/lineage.py

```diff
@@ -182,64 +186,121 @@ def _build_graphql_query(
                 )
             )
 
-        object_domain = self._user_to_system_domain_map.get(
-            object_domain.lower(), object_domain
-        )
-
         parent_param = ""
-        query_object_name = object_name
-        if object_version:
-            parent_param = f', {_ObjectField.PARENT_NAME}:"{object_name}"'
-            query_object_name = object_version
+        if object_id:
+            object_key = "id"
+            query_object = object_id
+            if parent_id:
+                parent_param = f', parentId:"{parent_id}"'
+        else:
+            object_key = "name"
+            if object_domain == _UserDomain.FEATURE_VIEW:
+                if not object_version:
+                    raise ValueError(
+                        f"Version cant be empty for {_UserDomain.FEATURE_VIEW}"
+                    )
+                object_name = _DGQLQueryBuilder._get_feature_view_name(
+                    object_name, object_version
+                )
+                object_version = None
+            object_name = object_name.replace('"', '\\\\"')
+            query_object = object_name
+            if object_version:
+                object_version = object_version.replace('"', '\\\\"')
+                query_object = object_version
+                parent_param = f', parentName:"{object_name}"'
+
+        object_domain = _DGQLQueryBuilder.USER_TO_SYSTEM_DOMAIN_MAP.get(
+            object_domain.upper(), object_domain
+        )
 
-        query_template = '{{{nodeKey}({domainKey}: {domain}, {nameKey}:"{name}"{parent_param}) {{{edges}}}}}'
-        query = query_template.format(
+        query = _DGQLQueryBuilder.QUERY_TEMPLETE.format(
             nodeKey=_DGQLFields.NODE,
             domainKey=_ObjectField.DOMAIN,
-            nameKey=_ObjectField.NAME,
             domain=object_domain.upper(),
-            name=query_object_name,
+            object_key=object_key,
+            query_object=query_object,
             parent_param=parent_param,
             edges="".join(parts),
         )
-
         return f"select SYSTEM$DGQL('{query}')"
 
+    @staticmethod
+    def split_fully_qualified_name(name):
+        """
+        Splits the fully qualified name.
+        Pattern matches either a string enclosed in double quotes or a sequence of word characters.
+        """
+        parts = re.findall(r'"[^"]*"|\w+', name)
+        return parts
+
+    @staticmethod
+    def _get_feature_view_name(name: str, version: str) -> str:
+        """
+        Constructs feature view name.
+        """
+        parts = _DGQLQueryBuilder.split_fully_qualified_name(name)
+        if len(parts) < 3:
+            raise ValueError("Invalid object name: less than three parts")
+
+        # Unquote the 3rd part if it exists
+        name_part = parts[2].strip('"')
+
+        # Feature view name is case insensitive
+        feature_view_name = f"{name_part.upper()}${version}"
```

Comment:
is this true? customers cannot name a DT/feature view with quoted names in lower case?

Reply:
The feature view name is case insensitive. But the version is sensitive. To treat the dynamic table/view name which is a combination of both as case sensitive, we always keep the name is upper case.

Reply:
> customers cannot name a DT/feature view with quoted names in lower case?
I did a test, it turns out that they can. I have changed the logic.